app/controllers/teams/joins_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Teams::JoinsController < TeamsController
+  skip_before_action :find_team
+
+  def show
+    team = Team.find_by!(token: params[:team_token], url_join_allowed: true)
```

Comment:
Can you add a guard here so if someone is already a member of this team, they get redirected to the team page please.